git_machete/client.py

```diff
@@ -1530,18 +1535,20 @@ def handle_untracked_branch(self, new_remote: str, branch: str, is_called_from_t
         other_remote_choice = "o[ther-remote]" if can_pick_other_remote else ""
         remote_branch = f"{new_remote}/{branch}"
         if not self.__git.get_commit_sha_by_revision(remote_branch, prefix="refs/remotes/"):
-            choices = get_pretty_choices(*('y', 'N', 'q', 'yq', other_remote_choice) if is_called_from_traverse else ('y', 'q', other_remote_choice))
+            choices = get_pretty_choices(*('y', 'N', 'q', 'yq', other_remote_choice) if is_called_from_traverse else ('y', 'Q', other_remote_choice))
             ask_message = f"Push untracked branch {bold(branch)} to {bold(new_remote)}?" + choices
             ask_opt_yes_message = f"Pushing untracked branch {bold(branch)} to {bold(new_remote)}..."
             ans = self.ask_if(ask_message, ask_opt_yes_message,
                               override_answer=None if self.__git.cli_opts.opt_push_untracked else "N")
             if ans in ('y', 'yes', 'yq'):
                 self.__git.push(new_remote, branch)
-                if ans == 'yq':
+                if ans == 'yq' and is_called_from_traverse:
                     raise StopTraversal
                 self.flush_caches()
             elif can_pick_other_remote and ans in ('o', 'other'):
                 self.__pick_remote(branch, is_called_from_traverse)
+            elif not is_called_from_traverse and ans not in ('y', 'yes'):
```

Comment:
TBH this entire logic is pretty complex now, and there are probably small bugs lurking here and there (like, `yq` is in fact still allowed for `github checkout-pr`, but won't really lead to "quitting" as one might expect, but instead proceed exactly as `y` AFAICS)...

Reply:
I'd suggest this entire logic (probably even from line 1537 onwards?) into completely separate cases for `is_called_from_traverse` being `true` and `false`, that will be waaaay easier to analyse